widgetssdk/src/main/java/com/glia/widgets/engagement/end/EngagementCompletionController.kt

```diff
@@ -0,0 +1,127 @@
+package com.glia.widgets.engagement.end
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import com.glia.androidsdk.engagement.Survey
+import com.glia.widgets.GliaWidgets
+import com.glia.widgets.UiTheme
+import com.glia.widgets.call.CallActivity
+import com.glia.widgets.chat.ChatActivity
+import com.glia.widgets.di.Dependencies
+import com.glia.widgets.engagement.EngagementRepository
+import com.glia.widgets.engagement.EngagementStateUseCase
+import com.glia.widgets.engagement.ReleaseResourcesUseCase
+import com.glia.widgets.engagement.SurveyUseCase
+import com.glia.widgets.helper.OneTimeEvent
+import com.glia.widgets.helper.getFullHybridTheme
+import com.glia.widgets.helper.isGlia
+import com.glia.widgets.helper.wrapWithMaterialThemeOverlay
+import io.reactivex.Flowable
+import io.reactivex.processors.PublishProcessor
+import java.lang.ref.WeakReference
+
+internal interface EngagementCompletionController {
+
+    val state: Flowable<State>
+    fun captureTheme(activity: Activity)
+    fun onActivityResumed(activity: Activity)
+    fun onActivityPaused()
+
+    sealed interface State {
+        object Skip : State
+        object ReleaseUi : State
+        data class ShowSurvey(val survey: Survey, val activity: Activity, val uiTheme: UiTheme) : State
+        data class ShowOperatorEndedEngagementDialog(val themedContext: Context, val uiTheme: UiTheme) : State
+        data class LaunchDialogHolderActivity(val activity: Activity) : State
```

Comment:
Could you please explain or even add comments to the `Skip`, `ReleaseUi` and `LaunchDialogHolderActivity` states?